tests/functional/aws-node-sdk/test/bucket/getCorsResponse.js

```diff
@@ -0,0 +1,1083 @@
+            body.push(chunk);
+        });
+        res.on('error', err => {
+            process.stdout.write('err receiving response');
+            return callback(err);
+        });
+        res.on('end', () => {
+            const total = body.join('');
+            if (code) {
+                const message = code === 200 ? '' : `<Code>${code}</Code>`;
+                assert(total.indexOf(message) > -1, `Expected ${message}`);
+                assert.deepEqual(res.statusCode, statusCode[code],
+                `status code expected: ${statusCode[code]}`);
+            }
+            if (headersResponse) {
+                Object.keys(headersResponse).forEach(key => {
+                    assert.deepEqual(res.headers[key], headersResponse[key],
+                      `error header: ${key}`);
+                });
+            } else {
+                ['access-control-allow-origin',
+                'access-control-allow-methods',
+                'access-control-allow-credentials',
+                'vary'].forEach(key => {
+                    assert.strictEqual(res.headers[key], undefined,
+                        `Error: ${key} should not have value`);
+                });
+            }
+            return callback();
+        });
+    });
+
+    req.on('error', err => {
+        process.stdout.write('err sending request');
+        return callback(err);
+    });
+    req.end();
+}
+
+describe('OPTIONS request without bucket', () => {
+    it('should respond no such bucket if bucket does not exist', done => {
+        const headers = {
+            Origin: 'http://www.allowedwebsite.com',
+        };
+        methodRequest({ method: 'GET', headers, code: 'NoSuchBucket',
+        headersResponse: null }, done);
+    });
+    it('should return BadRequest for OPTIONS request without origin', done => {
+        const headers = {};
+        methodRequest({ method: 'OPTIONS', headers, code: 'BadRequest',
+        headersResponse: null }, done);
+    });
+    it('should return BadRequest for OPTIONS request without ' +
+    'Access-Control-Request-Method', done => {
+        const headers = {
+            Origin: 'http://www.allowedwebsite.com',
+        };
+        methodRequest({ method: 'OPTIONS', headers, code: 'BadRequest',
+        headersResponse: null }, done);
+    });
+});
+
+describe('OPTIONS requests with existing bucket', () => {
+    beforeEach(done => {
+        s3.createBucket({ Bucket: bucket, ACL: 'public-read' }, err => {
+            _waitForAWS(done, err);
+        });
+    });
+    afterEach(done => {
+        s3.deleteBucket({ Bucket: bucket }, err => {
+            _waitForAWS(done, err);
+        });
+    });
+
+    it('should allow GET without CORS', done => {
+        const headers = {
+            Origin: 'http://www.allowedwebsite.com',
+        };
+        methodRequest({ method: 'GET', headers, code: 200,
+        headersResponse: null }, done);
+    });
+    it('should allow HEAD without CORS', done => {
```

Comment:
should allow HEAD request on bucket without cors configuration even if origin header sent